genesyscloud/responsemanagement_responseasset/resource_genesyscloud_responsemanagement_responseasset_utils.go

```diff
@@ -0,0 +1,48 @@
+package responsemanagement_responseasset
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	"terraform-provider-genesyscloud/genesyscloud/util/files"
+)
+
+func responsemanagementResponseassetResolver(responseAssetId, exportDirectory, subDirectory string, configMap map[string]interface{}, meta interface{}) error {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getRespManagementRespAssetProxy(sdkConfig)
+
+	fileName := fmt.Sprintf("asset-%s.jpeg", responseAssetId)
+
+	fullPath := path.Join(exportDirectory, subDirectory)
+	if err := os.MkdirAll(fullPath, os.ModePerm); err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	data, _, err := proxy.getRespManagementRespAssetById(ctx, responseAssetId)
+	if err != nil {
+		return err
+	}
+
+	log.Println("Details: ", fullPath, fileName, subDirectory)
+	if err := files.DownloadExportFile(fullPath, fileName, *data.ContentLocation); err != nil {
+		return err
+	}
+
+	configMap["filepath"] = path.Join(subDirectory, fileName)
```

Comment:
Here we should be updating filename, since it is actually used in the api call to create the asset